pkg/controllers/plugin/pluginpreset_controller.go

```diff
@@ -218,6 +218,47 @@ func (r *PluginPresetReconciler) reconcilePluginPreset(ctx context.Context, pres
 	return skippedCondition, failedCondition, utilerrors.NewAggregate(allErrs)
 }
 
+func shouldSkipPlugin(plugin *greenhousev1alpha1.Plugin, preset *greenhousev1alpha1.PluginPreset) bool {
+	if plugin.Labels[greenhouseapis.LabelKeyPluginPreset] != preset.Name {
+		return true
+	}
+
+	if preset.Spec.Plugin.PluginDefinition != plugin.Spec.PluginDefinition {
+		return false
+	}
+
+	for _, presetOptionValue := range preset.Spec.Plugin.OptionValues {
+		if !containsPluginOption(plugin.Spec.OptionValues, presetOptionValue) {
+			return false
+		}
+	}
+
+	return len(preset.Spec.Plugin.OptionValues) >= countWithoutGlobalOption(plugin.Spec.OptionValues)
```

Comment:
This would not take `PluginDefinition.Options` into account that transport a `default` value to the `Plugin.OptionValues`, right? We are only excluding by prefix `global.greenhouse`...
Can you add a test case for a `PluginDefinition` with a defaulted `Option`?

Reply:
I think we should only care on the PluginPreset side that the values part of the PluginPreset Spec are also set on the Plugin. We can ignore everything else imo. The other values are set by the Webhooks and are not a concern for the PresetController

Reply:
Meaning we can loose the `len()` check, right?

Reply:
Without that check the test fails [here](https://github.com/cloudoperators/greenhouse/blob/2c1fe55b5214a20b3b80f2f0af1193e408ce63b5/pkg/controllers/plugin/pluginpreset_controller_test.go#L131).

Reply:
The assumption that everything besides the PluginPreset spec can be ignored was wrong. I had not considered the case where the managed Plugin was changed.
The failing testcase checks this condition that a PluginOptionValue added to the managed Plugin is reverted by the PluginPreset controller.

So it should not skip the Plugin if there are PluginOptionValues that are neither declared on the PluginPreset nor on the PluginDefinition.

Reply:
Done